srcCpp/RTIDDSImpl.cxx

```diff
@@ -1020,61 +1028,61 @@ class RTIDynamicDataPublisher : public IMessagingWriter
     bool Send(const TestMessage &message, bool isCftWildCardKey)
     {
         DDS_ReturnCode_t retcode;
+        DDS_Octet key_octets[KEY_SIZE];
         long key = 0;
 
-        data.clear_all_members();
-
+        if (_last_message_size != message.size) {
+            //Cannot use data.clear_member("bind_data") because:
+            //DDS_DynamicData_clear_member:unsupported for non-sparse types
+            data.clear_all_members();
+            retcode = data.set_octet_array(
```

Comment:
Check compatibility non-dynamic <-> dynamic

Reply:
What is the conclusion on this? @softwarejimenez?

Reply:
I have run a couple of examples about compatibility of non-dynamic <-> dynamic and it works fine using set_octect_array

Reply:
use set_seq

Reply:
Done